tracer/src/Datadog.Trace.Tools.Runner/Checks/AgentConnectivityCheck.cs

```diff
@@ -59,11 +66,25 @@ public static async Task<bool> Run(string url)
             }
             catch (Exception ex)
             {
-                Utils.WriteError(ErrorDetectingAgent(url, ex.Message));
+                Utils.WriteError(ErrorDetectingAgent(settings.AgentUri.ToString(), ex.Message));
                 return false;
             }
 
             return true;
         }
+
+        private static void DisplayInfoMessage(ImmutableExporterSettings settings)
+        {
+            var transport = "HTTP";
+            var endpoint = settings.AgentUri?.ToString();
+
+            if (settings.TracesTransport == TracesTransportType.UnixDomainSocket)
+            {
+                transport = "domain sockets";
+                endpoint = settings.TracesUnixDomainSocketPath;
+            }
+
+            AnsiConsole.WriteLine(ConnectToEndpointFormat(endpoint!, transport));
```

Comment:
You're saying `endpoint` can be null above (if `settings.AgentUri` is null), then flagging it as not null here (`!`). Is this correct?

Reply:
`AgentUri` should only be null when `TracesTransport` is UnixDomainSocket

Reply:
But I agree the way it's written makes it confusing 🤔 gonna refactor it